src/components/Api.js

```diff
@@ -0,0 +1,101 @@
+export default class Api {
+  constructor(url) {
+    this._baseUrl = url.baseUrl;
+    this._header = url.headers;
+  }
+
+ _checkRequest(res) {
+   if(res.ok) {
+    return res.json();
+   } else {
+    return Promise.reject(`Error: ${res.status}`);
+   }
+ }
+
+
+   // methods for working with the API =========================================
+
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, {
+      headers: this._header,
+    })
+    .then(this._checkRequest);
```

Comment:
**Could be improved**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkRequest)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.